src/app/parcels/parcelsTable/fetchParcelTableData.ts

```diff
@@ -280,3 +281,21 @@ export const getClientIdForParcel = async (parcelId: string): Promise<FetchClien
 
     return { clientId: clientIdData.client_id, error: null };
 };
+
+export const getParcelsWithEvent = async (
+    targetEventName: StatusType,
+    parcelIds: string[]
+): Promise<GetDbParcelDataResult> => {
+    const { data, error } = await supabase
+        .from("parcels_plus")
+        .select("*, events!inner(new_parcel_status)")
```

Comment:
I'm not super familiar with this syntax but do we need to select * or can we just select parcel id, postcode and event status? (assuming those are not the only fields returned by the query)

Reply:
I'm also not too familiar, but when I tried to just select some of the fields, there was a type error because it no longer matched the full schema of the `parcels_plus` view. I think other queries generally just used a `select *` so I copied that.

Reply:
I would suggest making your own type using the `parcels_plus` schema and `Pick` to avoid having to fetch so much data. Then we can use that type in the return type of the function

Reply:
Thanks! I realised the only field we actually needed was the postcode, so I've changed it to just fetch that now, and moved the uniqueness stuff into the same function now that it's just getting postcodes.